cloudprovider/aws/aws_provider.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/rancher/agent/utilities/config"
+	"os"
+	"path"
+	"time"
+)
+
+const (
+	regionLabel           = "ec2.region"
+	availabilityZoneLabel = "ec2.availability_zone"
+	location              = "aws"
+	infoFile              = "info.json"
+	tempFile              = "temp.json"
+)
+
+type Info struct {
+	Label map[string]string `json:"label,omitempty"`
+}
+
+func CallProvider() {
+	infoPath := path.Join(config.StateDir(), location, infoFile)
+	tempPath := path.Join(config.StateDir(), location, tempFile)
+	select {
```

Comment:
I'm pretty sure you've structure this select statement + for loop incorrectly.
This will immediately drop into the default section and then loop _forever_. The timeout logic will never get hit.